packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -3846,3 +3846,99 @@ export class TerminalQuickFixOpener {
     constructor(uri: theia.Uri) { }
 }
 
+// #region Chat
+export class ChatRequestTurn {
+    readonly prompt: string;
+    readonly participant: string;
+    readonly command?: string;
+    readonly references: theia.ChatPromptReference[];
+    private constructor(prompt: string, command: string | undefined, references: theia.ChatPromptReference[], participant: string) {
+        this.prompt = prompt;
+        this.command = command;
+        this.participant = participant;
+        this.references = references;
+    };
+}
+
+export class ChatResponseTurn {
+    readonly command?: string;
+
+    private constructor(public readonly response: ReadonlyArray<theia.ChatResponseMarkdownPart | theia.ChatResponseFileTreePart | theia.ChatResponseAnchorPart
+        | theia.ChatResponseCommandButtonPart>, public readonly result: theia.ChatResult, public readonly participant: string) { }
+}
+
+export class ChatResponseAnchorPart {
+    constructor(public value: URI | Location, public title?: string) { }
+}
+
+export class ChatResponseProgressPart {
+    constructor(public value: string) { }
+}
+
+export class ChatResponseReferencePart {
+    constructor(public value: URI | theia.Location, public iconPath?: URI | ThemeIcon | {
+        light: URI;
+        dark: URI;
+    }) { }
+}
+export class ChatResponseCommandButtonPart {
+    constructor(public value: theia.Command) { }
+}
+
+export class ChatResponseMarkdownPart {
+    constructor(public value: string | theia.MarkdownString) { }
+}
+
+export class ChatResponseFileTreePart {
+    constructor(public value: theia.ChatResponseFileTree[], public baseUri: URI) { }
+}
+
+export type ChatResponsePart = ChatResponseMarkdownPart | ChatResponseFileTreePart | ChatResponseAnchorPart
+| ChatResponseProgressPart | ChatResponseReferencePart | ChatResponseCommandButtonPart;
+
+export enum ChatResultFeedbackKind {
+    Unhelpful = 0,
+    Helpful = 1,
+}
+
+export enum LanguageModelChatMessageRole {
+    User = 1,
+    Assistant = 2
+}
+
+export class LanguageModelChatMessage {
+
+    static User(content: string, name?: string): LanguageModelChatMessage {
+        return new LanguageModelChatMessage(LanguageModelChatMessageRole.User, content, name);
+    }
+
+    static Assistant(content: string, name?: string): LanguageModelChatMessage {
+        return new LanguageModelChatMessage(LanguageModelChatMessageRole.Assistant, content, name);
+    }
+
+    constructor(public role: LanguageModelChatMessageRole, public content: string, public name?: string) { }
+}
+
+export class LanguageModelError extends Error {
+
+    static NoPermissions(message?: string): LanguageModelError {
+        return new LanguageModelError(message, LanguageModelError.NotFound.name);
```

Comment:
This seems strange when reading (https://code.visualstudio.com/api/references/vscode-api#LanguageModelError). What should the `Error.name` and `code` properties be when we construct a `LanguageModelError` with the `NoPermissions` static method?` Right now, it would be `code === NotFound` and `name === `LanguageModelError`, right? What am I missing?

Reply:
I fixed this part. The LanguageModelError NoPermissions/NotFound/Blocked should now have the correct code.

BTW, due to Typescript version used, the 'cause' property used in VS code in the [constructor](https://github.com/microsoft/vscode/blob/ca8cb6f637fc5ac820b581eb4115bc3917ba7548/src/vs/workbench/api/common/extHostTypes.ts#L4568) is not available yet (es2022). Is there a way to track these issues, as we do for the monaco uplift annotations? should we introduce a typescript-uplift annotation?

Reply:
I doubt this is an issue of the typescript version: we are using 5.4 at the moment, which is fairly new. I figure the problem is with the compilation target, right?

Reply:
yes, the problem is the target